hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/StreamWriteFunction.java

```diff
@@ -385,16 +393,24 @@ private String getBucketID(HoodieRecord<?> record) {
    * @param value HoodieRecord
    */
   protected void bufferRecord(HoodieRecord<?> value) {
+    writeMetrics.markRecordIn();
     final String bucketID = getBucketID(value);
 
     DataBucket bucket = this.buckets.computeIfAbsent(bucketID,
-        k -> new DataBucket(this.config.getDouble(FlinkOptions.WRITE_BATCH_SIZE), value));
+        k -> {
+          // create a new bucket and update metrics
+          writeMetrics.increaseNumOfOpenHandle();
+          return new DataBucket(this.config.getDouble(FlinkOptions.WRITE_BATCH_SIZE), value);
```

Comment:
This is just in-memory bucket, not opening file handle.

Reply:
Yes, for stream write we will only create the handle when flushing buckets. To avoid misleading, remove metrics here since for stream write, the metrics `numOfFilesWritten` is enough.